### ISSUE TITLE
All todos disappeared, can't add new ones

### ISSUE THREAD

**Issue by gregorycerna** [HabitRPG/habitica#1436]
Multiple copies of the same task showed up on the mobile app (Android) when I added a task on the main website. I discovered this after trying and failing to add a second task. When I tried to delete some of the duplicates via the mobile app, ALL of my todos were deleted, including my completed list. Unfortunately, I can't add any more todos now, either from the main website, the beta website, or the mobile app.

**Comment by lefnire** [comment]
yowzer :/ what's your uuid, I might not be able to jump in soon - but this obviously needs to get fixed

**Comment by gregorycerna** [comment]
My username is gregcerna and my user ID is 99179b63-9253-497a-a11e-70e83af824e. Let me know if there's anything else I can do to help.

**Comment by lefnire** [comment]
very interesting. you do indeed have duplicate task UUIDs. all the todos are still intact, it's just that Angular doens't want to render the duplicates, so it cuts out early the moment it encounters a dupe. Can you go to the [OLD website](http://old.habitrpg.com), just visit it and let it load, and see if it clears up the issue? This issue shouldn't happen again one we kill of the main site, but we'll want to clear you up this time. The old site has a "duplicates sanitization" method that's run on load (since it was so notorious for creating duplicates), which doesn't exist on the new site or mobile since it's not needed.

**Comment by gregorycerna** [comment]
Yea, that did the trick. Thanks for your help!

**Comment by lefnire** [comment]
Kk, keep an close eye on this. I'm gonna close this ticket, if it happens again _on beta or mobile_ (or on master, once we push that in a few days) - then re-open

**Comment by eri401** [comment]
This just happened to me.  I just downloaded the ios app and was just checking it out.

Then I went to my regular online version.

Like the last user, multiple copies of the same task showed up on my DESKTOP.  When I began to clear them out, I refreshed the page and suddenly all tasks were deleted and similarly, my completed list was cleared out as well.   Now I cannot add anything.  

I went to the main website and it did NOT clear up the issue for me.  I closed the iphone app and logged out of my online version.  Then I refreshed and logged back in but it still is not working. 

Not sure if it'll help but 
User Name: jelliajamb
User ID: d50ece58-05fd-4c52-a4d6-089e19ac5bf9
Chrome browser, ios app

**Comment by lefnire** [comment]
I'll work on a temporary fix tonight. In the meantime @eri401 , visit old.habitrpg.com _once_ to re-calibrate, and then don't touch it again if at all possible. What's happening is Derby created all sorts of duplicates in the database. y'all didn't know since I was running around sanitizing it on the backend. Well now we're on angular + express, which don't create dupes and therefore don't need sanitization, I removed sanitization. I suppose I can't do this until even old.habit's dead (so i'll add that back in).

In the mean time, hit old.habit to sanitize, then avoid it afterwards as it's what's causing the dupes. why does everything disappear on mobile / new web? Angular doesn't like duplicates, so it just doesn't show them. They're still there, just invisible until dupes are reconciled.

Unfortunately, I don't know what will be the case with all the delete operations you clicked to try to get back to good. Could have borked things, I have a db backup we can restore you to if that's the case.

**Comment by eri401** [comment]
Once I went to old.habitrpg and logged in, I saw all my ToDos again! Including all the ones I recently tried to add while the new site wasn't working.

Then I went back to the new main page and they are all back. It also seems like I'm now able to add ToDos again.  I also checked my ios app and after syncing, it seems that is also working (I can add ToDos and also see what is added on the computer). 

Now that everything is in order, I won't bother with old.habitrpg again.  

Excited and happy that you guys are making progress.  Bugs aside, this still my favorite productivity site.
